pages/featured-articles/[slug].js

```diff
@@ -0,0 +1,80 @@
+// @author Charles Liu
+
+import { useRouter } from 'next/router'
+import Image from 'next/image'
+import Container from 'react-bootstrap/Container'
+import Row from 'react-bootstrap/Row'
+import Col from 'react-bootstrap/Col'
+import ReactMarkdown from 'react-markdown'
+import Moment from 'react-moment'
+
+import { fetchAPI } from '../../lib/api'
+import { getStrapiMedia } from '../../lib/media'
+import Banner from '../../components/shared/Banner'
+import styles from '../../styles/Article.module.css'
+
+function Article({ article }) {
+  return (
+    <>
+      <Banner url={getStrapiMedia(article.image[0])} title={article.title} />
+      <Container>
+        <Row className="mt-4">
+          <h1>{article.title}</h1>
+        </Row>
+        {article.subtitle && (
+          <Row>
+            <h3 className="text-secondary font-italic">{article.subtitle}</h3>
+          </Row>
+        )}
+        <Row>
+          <Col xs={12} md={4} className="p-0">
+            <div className={styles.authorContainer}>
+              <div className={styles.avatar}>
+                <Image
+                  src={getStrapiMedia(article.author.image)}
+                  layout="fill"
+                  objectFit="cover"
+                  objectPosition="center center"
+                />
+              </div>
+              <div className={styles.authorMeta}>
+                <p>{article.author.name}</p>
+                <Moment className="text-secondary" format="MMM Do YYYY">
+                  {article.published_at}
+                </Moment>
+              </div>
+            </div>
+          </Col>
+          <Col xs={12} md={8} className={styles.socialMedia}>
+            Social Media Icons
+          </Col>
+        </Row>
+        <Row className="mt-4 py-4">
+          <ReactMarkdown children={article.content} />
+        </Row>
+      </Container>
+    </>
+  )
+}
+
+export async function getStaticPaths() {
+  const articles = await fetchAPI('/articles')
+  const paths = articles.map((article) => ({
+    params: {
+      slug: article.slug,
+    },
+  }))
+  return {
+    paths,
+    fallback: false,
+  }
+}
+
+export async function getStaticProps({ params }) {
+  const articles = await fetchAPI(`/articles?slug=${params.slug}`)
+  return {
+    props: { article: articles[0] },
+  }
```

Comment:
Why are you always indexing the first element in `articles`? asking cuz idk how it works

Reply:
it returns an array of all articles with the slug. We just want the first one (which is usually the only one).